src/PrestaShopBundle/Entity/Repository/StockRepository.php

```diff
@@ -129,7 +129,7 @@ public function updateStock(Movement $movement, $syncStock = true)
                     $product,
                     $productIdentity->getCombinationId(),
                     $delta,
-                    $this->contextAdapter->getContext()->shop->id,
+                    $this->getCurrentShop()->id,
```

Comment:
Possibly yes, and it kinda bothers me :)
- current shop instance is stored in context, and its ID property is required here
- shop class also owns a static property `$context_id_shop` which is returned by `$shop->getContextualShopId()`. This static value is refreshed by `Shop::intialize()`... very confusing indeed.

I kept both values rigorously splitted to prevent any risk of edge case. But I rename method `StockRepository::getCurrentShopId()` to `getCurrentShopId()`, because it's only a shortcut to `$shop->getContextualShopId()`.

Reply:
magic !

Reply:
black magic :)

Reply:
Yeah it's safer to do this Both values should, theoretically, be synced But in case they are not let's keep the initial behaviour for now

Reply:
[Paranoid mode enabled] ;)